### PR TITLE
Introduce opt-in WooCommerce rebranding

### PR DESCRIPTION
**Pull request by Konamiman**
### Changes proposed in this Pull Request:

This pull request introduces the styling changes required to conform to the WooCommece rebranding to be rolled out in the following months. The changes are hidden by default, to acivate them one of the following snippets is required:

* `define('WOOCOMMERCE_USE_NEW_BRANDING', true;` - for example in `wp-config.php`.
* `add_filter('woocommerce_use_new_branding', '__return_true');` - for example using [a snippets extension](https://wordpress.org/plugins/code-snippets/).

A new `BrandingController` class and a few additional scss files are added, but these are temporary and only in order to be able to switch between the old and new branding; these files will be removed in a later release of WooCommerce.

The implemented changes are:

1. New icon in the admin dashboard:

image
image

2. New default color in emails:

image
image

**NOTE!** What changes is the _default_ value for the email color setting only, the currently configured color won't change automatically (with the changes of this pull request).

3. New primary color in the admin area:

image
image

### How to test the changes in this Pull Request:

Prepare a site with WooCommerce and:

- One product with reviews.
- One order, created either via checkout in the shop or directly in the admin area.

Also add the following snippet to the site:

```php
add_action( 'admin_init', function() {
    WC_Admin_Notices::add_notice( 'redirect_download_method' );
} );
```

1. Activate the new branding by either creating a `WOOCOMMERCE_USE_NEW_BRANDING` constant that returns `true`, or having the `woocommerce_use_new_branding` filter return true, as explained above.

2. Load the admin area of the site. Verify that the left menu shows the new icon in the "WooCommerce" entry, both when the menu is collapsed and expanded.

3. Go to Dashboard - Home (the main WordPress dahsboard, not the WooCommerce one). Verify that the "Redirect only" notice that appears at the top and the stars in "WooCommerce Recent Reviews" use the new color, `#720EEC`; you can use the inspector in the browser developer tools for that, e.g. for the notice:

image

**NOTE!** In other parts of the admin area (e.g. Products) you may see that the notice uses the default WordPress coloring instead. That's because a `--wp-admin-theme-color` CSS variable exists, but if you use the browser inspector you should see that the fallback color to use when that variable doesn't exist is the new one:

image

4. Go to WooCommerce- Settings - Email. Scroll down to the general settings for emails. You will see that the value for "Base color" hasn't changed, but if you click on the "?" icon the tooltip will show the new color as the default for the setting.

Here you can manually change the value to the new default (`#720EEC`), save, and send a test email (e.g. open the details of an order in the admin area, then "Order actions" - "Resend new order notification"). You can use [an email logger extension](https://wordpress.org/plugins/email-log/) if you don't want to send actual emails.

An alternative way to verify that the new default is actually a default could be:

* Install WordPress without WooCommerce.
* Add the constant or the filter to activate the new branding.
* Install WooCommerce from a zip file generated from this branch.
* Verify that the value of the base color setting for emails is the new default value.

5. Repeat the previous steps without the constant or filter, verify that the old branding is restored (the old default primary color for the admin area and emails is `#7F54B3`).

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by tomalec** [comment]
WDYT of using numbers like `woocommerce_use_woo2.0_branding` in variable and file names? Instead of old/new. 
So, if we are updating colors somewhere in the future, we will not have to wonder how old is the "old" and won't introduce `woocommerce_use_new_new_branding`

**Comment by tomalec** [comment]
Also, maybe you would like to couple this PR with updating color studio for all the packages 
https://github.com/woocommerce/woocommerce/pull/53123

**Comment by Konamiman** [comment]
> WDYT of using numbers like woocommerce_use_woo2.0_branding in variable and file names? Instead of old/new.
So, if we are updating colors somewhere in the future, we will not have to wonder how old is the "old" and won't introduce woocommerce_use_new_new_branding

This naming is temporary, there will be "old" and "new" files only as long as we are testing the new branding. Once this is officially released everything will go "back to normal", with the previous filenames restored and with no `BrandingController` class (a separate PR will take care of that).

**Comment by Konamiman** [comment]
> Also, maybe you would like to couple this PR with updating color studio for all the packages

I think that the safest course of action here would be:

1. Merge this PR
2. Rebase https://github.com/woocommerce/woocommerce/pull/53123 from trunk
3. Update https://github.com/woocommerce/woocommerce/pull/53123 so that it applies the new colors only when `BradingController::use_new_branding` returns `true`

Does that make sense?

**Comment by tomalec** [comment]
> Does that make sense?

The plan makes sense.
However

> so that it applies the new colors only when `BradingController::use_new_branding` returns `true`

That would be a bit tricky, as it means we need to bundle both versions of the library, and switch it somehow according to the PHP function. I'm not sure if we have any precedent where we do similar stuff, not to reinvent the wheel.

**Comment by Konamiman** [comment]
> we need to bundle both versions of the library, and switch it somehow according to the PHP function

How does that work? Does the choice of colors at runtime depend on the version of the library used?

File: plugins/woocommerce/client/legacy/css/activation-old-branding.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * activation-old-branding.scss
+ * Extra activation styles for the old WooCommerce branding
+ */
+
+/**
+ * Styling begins
+ */
+div.woocommerce-message {
+	&.updated {
+		border-left-color: var(--wp-admin-theme-color, #7F54B3) !important;
```
Comment (naman03malhotra) [inline plugins/woocommerce/client/legacy/css/activation-old-branding.scss:11]: same comment as above ^

File: plugins/woocommerce/client/legacy/css/activation-new-branding.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ * activation-new-branding.scss
+ * Extra activation styles for the new WooCommerce branding
+ */
+
+/**
+ * Styling begins
+ */
+div.woocommerce-message {
+	&.updated {
+		border-left-color: var(--wp-admin-theme-color, #720EEC) !important;
```
Comment (naman03malhotra) [inline plugins/woocommerce/client/legacy/css/activation-new-branding.scss:11]: I am not sure if need to use `!important` here, as a good practice we should move away from using overrides.

File: plugins/woocommerce/src/Internal/BrandingController.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		return apply_filters( 'woocommerce_use_new_branding', $should_use );
+	}
+
+	/**
+	 * Get the default base color for the emails sent by WooCommerce.
+	 *
+	 * @return string
+	 */
+	public static function get_default_email_base_color() {
+		return self::use_new_branding() ? '#720eec' : '#7f54b3';
```
Comment (naman03malhotra) [inline plugins/woocommerce/src/Internal/BrandingController.php:40]: Nit: perhaps we should name `#720eec` and `#7f54b3` for readability.

$new_primary_branding_color = `#720eec`

**Review by naman03malhotra — COMMENTED** [review]
Looks good to me overall and tests as expected, nice work!

>Here you can manually change the value to the new default (#720EEC),

Is there a reason we are not defaulting to this value automatically?

✅  Verify that the left menu shows the new icon in the "WooCommerce" entry, both when the menu is collapsed and expanded.:
image

✅  Go to Dashboard - Home (the main WordPress dahsboard, not the WooCommerce one). Verify that the "Redirect only" notice that appears at the top and the stars in "WooCommerce Recent Reviews" use the new color, #720EEC; you can use the inspector in the browser developer tools for that, e.g. for the notice:

Screenshot 2024-12-04 at 8 13 29 PM
Screenshot 2024-12-04 at 8 13 34 PM


✅  Go to WooCommerce- Settings - Email. Scroll down to the general settings for emails. You will see that the value for "Base color" hasn't changed, but if you click on the "?" icon the tooltip will show the new color as the default for the setting.

image

✅ Repeat the previous steps without the constant or filter, verify that the old branding is restored (the old default primary color for the admin area and emails is #7F54B3

NOTE: Stars didn't reader for me for some reason, maybe that's specific to my setup.
image

**Review by naman03malhotra — APPROVED** [review]
Pre-approving: Only some minor comments needs to be addressed.

**Comment by Konamiman** [comment]
> >Here you can manually change the value to the new default (#720EEC),

> Is there a reason we are not defaulting to this value automatically?

Yes. This is a setting that gets set to a default value at WooCommerce install time, and the user can later change it to a different value.

What we are modifying here is the default setting _for new installs_, and informing the user (in the setting tooltip) of which is that new default; but there's no practical way of modifying the existing value, without overriding any possible user custom value, and while keeping the new branding under the feature flag.

What we'll probably do once the rebranding is rolled out for everybody (so without any feature flag) is to introduce a data migration that will change the value of the setting to the new default, but **only** if the current value of the setting is the old default (so that any user customization in place isn't lost).

File: plugins/woocommerce/client/legacy/css/activation-new-branding.scss (hunk 2 above)
Comment (Konamiman) [reply plugins/woocommerce/client/legacy/css/activation-new-branding.scss:11]: You're probably right, but that would be a change to address in a different pull request. In this one I prefer to keep the changes to a minimum, so this styling is simply copied and pasted verbatim from what already exists in `activation.scss`.

File: plugins/woocommerce/src/Internal/BrandingController.php (hunk 3 above)
Comment (Konamiman) [reply plugins/woocommerce/src/Internal/BrandingController.php:40]: That would be a sensible thing to do for long lived code, but keep in mind that this class is temporary: it's going away once the rebranding is rolled out.

**Comment by tomalec** [comment]
> How does that work? Does the choice of colors at runtime depend on the version of the library used?

The thing is:
The old version of library, 2.5.0, has the old branding colors, like https://github.com/Automattic/color-studio/blob/5623403c5350d02ee453c31c9ddec2dd3907a8db/dist/color-variables.scss#L166  has `#7f54b3` for WooCommerce purple
while `4.0.0`, has the new colors
https://github.com/Automattic/color-studio/blob/15208bcc6265c0a3d71dbeaec25dd8a94973f7b4/dist/color-variables.scss#L180 and `#720eec`.
The `color-studio` package is used by various packages within the repo,  some of which use variables to compile the CSS on build time.

https://github.com/woocommerce/woocommerce/blob/bb391ca8c4d6c2403e7976bb29da28c60309d041/packages/js/components/src/calendar/style.scss#L98:L98


I guess we cannot rebuild all packages after the setting switch.


Then, those components will still have the old branding color, even if someone toggles the switch.

**Comment by Konamiman** [comment]
@tomalec 

> I guess we cannot rebuild all packages after the setting switch.

Yes, that's a tricky one. I guess this is one case where we can't rely on the feature flag; instead, we'll need to prepare a PR where the package version is updated, and defer mergein it until the point release where the new branding will be finally rolled out.

**Pull request merged**